src/examples/desktop_opengl.cpp

```diff
@@ -127,40 +149,109 @@ namespace jsar::example
         return false;
 
       int samples = 4;
-      int opt;
-      while ((opt = getopt(argc, argv, "w:h:x:mn:s:")) != -1)
+
+      // Parse arguments manually to support long options
+      for (int i = 1; i < argc; i++)
       {
-        switch (opt)
+        string arg = argv[i];
+
+        if (arg == "--help")
+        {
+          help();
+          return false;
+        }
+        else if (arg == "--mono")
+        {
+          monoMode = true;
+        }
+        else if (arg == "--stereo")
         {
-        case 'w':
-          width = atoi(optarg);
-          break;
-        case 'h':
-          height = atoi(optarg);
-          break;
-        case 'x':
-          if (strcmp(optarg, "r") == 0)
-            xrEnabled = true;
-          break;
-        case 'm':
-          multiPass = true;
-          break;
-        case 'n':
-          nApps = atoi(optarg);
+          if (i + 1 >= argc)
```

Comment:
Consider the following cases:

* `--stereo -w 100`
* `--stereo https://foobar`

You can't just check if the index is out of the argument size, i think it's good to check if the next arg is "multipass" or "singlepass", if not then the default(singlepass) can be used and the index could be increased.